### DISCUSSION TITLE
[feature]: List of bitcoin JSON-RPC methods used by lnd

### DISCUSSION THREAD

**Discussion by basharov** [lightningnetwork/lnd#8126] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I am working on a proxy/loadbalancer app that would be placed between `bitcoind` and `lnd` connecting to it.

`bitcoind <> proxy <> lnd`.

Proxy will check some credentials like `rpcpassword` and `rpcuser` and if an RPC method name is correct, e.g. `getblockchaininfo` and pass it through to bitcoind then sending back the response. If the method is not correct, it will return an error.

I believe lnd won't use the [whole set of RPC methods](https://developer.bitcoin.org/reference/rpc/) available in `bitcoind` so I would allow only those that are used by lnd and forbid all the others.

In order to do that I would be happy to get a clue where I can look for those used by lnd.

**Describe the solution you'd like**
Have some advise on how to look for RPC methods that lnd uses to communicate with bitcoin core

**Describe alternatives you've considered**
I looked through the code and found some methods but I am not sure those are all I found.

**Comment by guggero** [comment, marked as answer]
A good place to start is probably [`BitcoinClient`](https://github.com/btcsuite/btcwallet/blob/55b02c6a7514a645630da4fadcf5a85b46a96a64/chain/bitcoind_client.go#L30) in the wallet library. That is certainly not an exhaustive list and I'm not sure if `lnd` calls all those methods. But I'd say that would cover around 80 to 90% of the calls.
Another place to look is the [`BitcoindNotifier`](https://github.com/lightningnetwork/lnd/blob/66b8700c0b8de452508d26cb2c05bad91aee4f98/chainntnfs/bitcoindnotify/bitcoind.go#L40) in `lnd`.

**Comment by saubyk** [comment]
Converting this issue to a discussion, as I don't think we have any action item to address here.
